src/main/io/osd.c

```diff
@@ -518,17 +518,20 @@ static bool osdDrawSingleElement(uint8_t item)
         }
 
     case OSD_CRAFT_NAME:
+        // This does not strictly support iterative updating if the craft name changes at run time. But since the craft name is not supposed to be changing this should not matter, and blanking the entire length of the craft name string on update will make it impossible to configure elements to be displayed on the right hand side of the craft name.
+        //TODO: When iterative updating is implemented, change this so the craft name is only printed once whenever the OSD 'flight' screen is entered.
+
         if (strlen(pilotConfig()->name) == 0) {
             strcpy(buff, "CRAFT_NAME");
         } else {
             for (unsigned int i = 0; i < MAX_NAME_LENGTH; i++) {
```

Comment:
Since we need a terminator anyway, how about:
```
            unsigned int I;
            for (i = 0; i < MAX_NAME_LENGTH; i++) {
                if (pilotConfig()->name[i] == 0) {
                    break;
                } else {
                    buff[i] = toupper((unsigned char)pilotConfig()->name[i]);
                }    
            }    
            buff[i] = '\0';
```

Reply:
Good point, that makes it more obvious compared to feeding `\0` through `toupper()`.

Reply:
You should not do `buff[i] = 0`. Instead do` buff[i] = SYM_BLANK` . Or even better, do nothing at all. No need for a terminated string in the buffer.

Reply:
Really? I thought `buff` is passed to `displayWrite` as a null terminated string.

Reply:
Ah, sorry. This was already discussed :)